### PR TITLE
Adds doGet() in LectureServlet.java

### PR DESCRIPTION
**Pull request by jaysaleh**
This pull request adds `doGet()` in `LectureServlet.java` which queries the database and responds with a `JSON` containing all the lectures in the database.

### CODE REVIEW

File: src/main/java/com/googleinterns/zoomtube/data/Lecture.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@AutoValue 
+public abstract class Lecture {
+  public abstract long id();
+  public abstract String lectureName();
+  public abstract String videoUrl();
+  public abstract String videoId();
+
+  /** 
+   * Creates a Lecture.
+   * @param id Id of object stored in database
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/data/Lecture.java:30]: I think you need a period after each @param description.

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 2]
```diff
@@ -63,20 +68,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+      String videoUrl = (String) lecture.getProperty(VIDEO_URL);
+      String videoId = (String) lecture.getProperty(VIDEO_ID);
+      
+      // Creates new Comment for JSON accessibility
+      Lecture newLecture = Lecture.create(id, lectureName, videoUrl, videoId);
+      lectures.add(newLecture);
+    }
+
+    Gson gson = new Gson();
+    response.setContentType("text/html");
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:92]:
```suggestion
    response.setContentType("application/json;");
```

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 3]
```diff
@@ -63,20 +67,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+    response.getWriter().println(gson.toJson(lectures));
   }
 
   /**
-   * Returns value with {@code name} from the {@code request} form. 
+   * Returns value with {@code name} from the {@code request} form.
    * If the {@code name} cannot be found, return {@code defaultValue}.
    * @param request Form sent by client
    * @param name {@code <input>} to read content of
+   * @param defaultValue Value to return if no content is found
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:101]: period here too

File: src/main/java/com/googleinterns/zoomtube/data/Lecture.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+/** Stores data for lectures added to site. */
+@AutoValue
+public abstract class Lecture {
+  public abstract long id();
+  public abstract String lectureName();
+  public abstract String videoUrl();
+  public abstract String videoId();
+
+  /**
+   * Creates a Lecture.
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/data/Lecture.java:29]: https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs

There should be a newline between the summary and the block tags, I think
```suggestion
   * Creates a Lecture.
   *
```

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 5]
```diff
@@ -63,20 +67,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(lectureEntity);
-
-    // TODO: Add redirect link to lecture site.
+    response.sendRedirect("/index.html");
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:70]: Maybe this should be turned into a constant?  Also, we can redirect to just `/` to keep things cleaner

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 6]
```diff
@@ -63,20 +67,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+    Gson gson = new Gson();
+    response.setContentType("text/html");
+    response.getWriter().println(gson.toJson(lectures));
   }
 
   /**
-   * Returns value with {@code name} from the {@code request} form. 
+   * Returns value with {@code name} from the {@code request} form.
    * If the {@code name} cannot be found, return {@code defaultValue}.
    * @param request Form sent by client
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:99]: Newline before block tags

```suggestion
   *
   * @param request Form sent by client
```

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 7]
```diff
@@ -85,10 +107,11 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
-  
+
   /** Returns YouTube video ID for a given {@code videoUrl}. */
   private String getVideoId(String videoUrl) {
-    String pattern = "(?<=watch\\?v=|/videos/|embed\\/|youtu.be\\/|\\/v\\/|\\/e\\/|watch\\?v%3D|watch\\?feature=player_embedded&v=|%2Fvideos%2F|embed%\u200C\u200B2F|youtu.be%2F|%2Fv%2F)[^#\\&\\?\\n]*";
+    String pattern =
+        "(?<=watch\\?v=|/videos/|embed\\/|youtu.be\\/|\\/v\\/|\\/e\\/|watch\\?v%3D|watch\\?feature=player_embedded&v=|%2Fvideos%2F|embed%\u200C\u200B2F|youtu.be%2F|%2Fv%2F)[^#\\&\\?\\n]*";
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:114]: Consider making this a constant so you can have a comment explaining it

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 8]
```diff
@@ -85,10 +107,11 @@ private String getParameter(HttpServletRequest request, String name, String defa
...
     Pattern compiledPattern = Pattern.compile(pattern);
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:115]: You could also make the Pattern a field that's set in `init` using the regex constant so it isn't recompiled every request.

File: src/main/java/com/googleinterns/zoomtube/data/Lecture.java (hunk 1 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/data/Lecture.java:30]: fixed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 3 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:101]: removed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 5 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:70]: fixed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 7 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:114]: fixed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 8 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:115]: fixed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 6 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:99]: removed

File: src/main/java/com/googleinterns/zoomtube/data/Lecture.java (hunk 4 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/data/Lecture.java:29]: fixed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 2 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:92]: fixed

**Pull request merged**